client/src/screens/search-groups.screen.js

```diff
@@ -68,21 +69,37 @@ class Group extends Component {
   constructor(props) {
     super(props);
     this.myGroups = this.props.myGroups;
-    this.joinGroupQuery = this.props.joinGroupQuery.bind(this);
-    this.joinGroup = this.props.joinGroup.bind(this, this.props.group);
+    this.joinGroup = this.joinGroup.bind(this);
+    this.groupUpdate = this.props.groupUpdate.bind(this)
+    this.state = {
+      memberAlready: _.some(this.myGroups, g => g.id === this.props.group.id),
+    };
+
+  }
+
+  joinGroup() {
+    const { groupUpdate, group } = this.props;
+    groupUpdate({ groupId: group.id }).then((res) => {
+      this.setState({memberAlready:true });
```

Comment:
nice - but lets make sure this is actually needed - if we do it right we shouldn't need this because the apollo store update should cause a refresh (unchecked hypothesis)

Reply:
It cases the group object to re-render but it wont have any new data because the AllGroup object isnt re-rendering, so it doenst know its joined a new group